<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from iiwa14.urdf.xacro              | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!--|
    Copyright (C) 2019-2022 Learning Algorithms and Systems Laboratory, EPFL, Switzerland
    Authors:  Konstantinos Chatzilygeroudis (maintainer)
              Matthias Mayr
              Bernardo Fichera
    email:    costashatz@gmail.com
              matthias.mayr@cs.lth.se
              bernardo.fichera@epfl.ch
    Other contributors:
              Yoan Mollard (yoan@aubrune.eu)
              Walid Amanhoud (walid.amanhoud@epfl.ch)
    website:  lasa.epfl.ch

    This file is part of iiwa_ros.

    iiwa_ros is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    iiwa_ros is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

|-->
<robot name="iiwa14">
  <material name="Black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="Blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="Green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.4 0.4 0.4 1.0"/>
  </material>
  <material name="Orange">
    <color rgba="1.0 0.4235294117647059 0.0392156862745098 1.0"/>
  </material>
  <material name="Brown">
    <color rgba="0.8705882352941177 0.8117647058823529 0.7647058823529411 1.0"/>
  </material>
  <material name="Red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="White">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <!-- Fix to world just for testing -->
  <link name="world"/>
  <!--joint between {parent} and link_0-->
  <joint name="world_iiwa_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="base_link"/>
  </joint>
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.1 0 0.07"/>
      <mass value="5"/>
      <inertia ixx="0.05" ixy="0" ixz="0" iyy="0.06" iyz="0" izz="0.03"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/iiwa14/link_0.stl"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>
  <!-- joint between link_0 and link_1 -->
  <joint name="joint_a1" type="revolute">
    <parent link="base_link"/>
    <child link="iiwa_link_1"/>
    <origin rpy="0 0 0" xyz="0 0 0.1575"/>
    <axis xyz="0 0 1"/>
    <limit effort="320" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="1.4835298641951802"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-2.8187067419708423" soft_upper_limit="2.8187067419708423"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="iiwa_link_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0 -0.03 0.12"/>
      <mass value="4"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.09" iyz="0" izz="0.02"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/iiwa14/link_1.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
  </link>
  <!-- joint between link_1 and link_2 -->
  <joint name="joint_a2" type="revolute">
    <parent link="iiwa_link_1"/>
    <child link="iiwa_link_2"/>
    <origin rpy="1.5707963267948966   0 3.141592653589793" xyz="0 0 0.2025"/>
    <axis xyz="0 0 1"/>
    <limit effort="320" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="1.4835298641951802"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-1.9896753472735356" soft_upper_limit="1.9896753472735356"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="iiwa_link_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0003 0.059 0.042"/>
      <mass value="4"/>
      <inertia ixx="0.05" ixy="0" ixz="0" iyy="0.018" iyz="0" izz="0.044"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/iiwa14/link_2.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
  </link>
  <!-- joint between link_2 and link_3 -->
  <joint name="joint_a3" type="revolute">
    <parent link="iiwa_link_2"/>
    <child link="iiwa_link_3"/>
    <origin rpy="1.5707963267948966 0 3.141592653589793" xyz="0 0.2045 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="176" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="1.7453292519943295"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-2.8187067419708423" soft_upper_limit="2.8187067419708423"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="iiwa_link_3">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0.03 0.13"/>
      <mass value="3"/>
      <inertia ixx="0.08" ixy="0" ixz="0" iyy="0.075" iyz="0" izz="0.01"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/iiwa14/link_3.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
  </link>
  <!-- joint between link_3 and link_4 -->
  <joint name="joint_a4" type="revolute">
    <parent link="iiwa_link_3"/>
    <child link="iiwa_link_4"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 0 0.2155"/>
    <axis xyz="0 0 1"/>
    <limit effort="176" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="1.3089969389957472"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-1.9896753472735356" soft_upper_limit="1.9896753472735356"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="iiwa_link_4">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0.067 0.034"/>
      <mass value="2.7"/>
      <inertia ixx="0.03" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.029"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/iiwa14/link_4.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
  </link>
  <!-- joint between link_4 and link_5 -->
  <joint name="joint_a5" type="revolute">
    <parent link="iiwa_link_4"/>
    <child link="iiwa_link_5"/>
    <origin rpy="-1.5707963267948966 3.141592653589793 0" xyz="0 0.1845 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="110" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="2.2689280275926285"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-2.8187067419708423" soft_upper_limit="2.8187067419708423"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="iiwa_link_5">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0001 0.021 0.076"/>
      <mass value="1.7"/>
      <inertia ixx="0.02" ixy="0" ixz="0" iyy="0.018" iyz="0" izz="0.005"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/iiwa14/link_5.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
  </link>
  <!-- joint between link_5 and link_6 -->
  <joint name="joint_a6" type="revolute">
    <parent link="iiwa_link_5"/>
    <child link="iiwa_link_6"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 0 0.2155"/>
    <axis xyz="0 0 1"/>
    <limit effort="40" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="2.356194490192345"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-1.9896753472735356" soft_upper_limit="1.9896753472735356"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="iiwa_link_6">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0.0006 0.0004"/>
      <mass value="1.8"/>
      <inertia ixx="0.005" ixy="0" ixz="0" iyy="0.0036" iyz="0" izz="0.0047"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/iiwa14/link_6.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
  </link>
  <!-- joint between link_6 and link_7 -->
  <joint name="joint_a7" type="revolute">
    <parent link="iiwa_link_6"/>
    <child link="iiwa_link_7"/>
    <origin rpy="-1.5707963267948966 3.141592653589793 0" xyz="0 0.081 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="40" lower="-3.0543261909900763" upper="3.0543261909900763" velocity="2.356194490192345"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-2.901609881440573" soft_upper_limit="2.901609881440573"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="iiwa_link_7">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.02"/>
      <mass value="0.3"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/iiwa14/link_7.stl"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>
  <joint name="iiwa_joint_ee" type="fixed">
    <parent link="iiwa_link_7"/>
    <child link="iiwa_link_ee"/>
    <origin rpy="0 0 0" xyz="0 0 0.045"/>
  </joint>
  <link name="iiwa_link_ee">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1e-6"/>
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
    </inertial>
  </link>
  <link name="ball_link">
    <visual>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>

      <geometry>
        <sphere radius="0.005"/>
      </geometry>
    </visual>
  </link>
  <joint name="ball_joint" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="-6.92119286580406e-05 -0.00018859822675517448 0.03542481345977316" />
    
    <parent link="iiwa_link_ee" />
    <child link="ball_link" />
  </joint>
  <!-- Load Gazebo lib and set the robot namespace -->
</robot>

